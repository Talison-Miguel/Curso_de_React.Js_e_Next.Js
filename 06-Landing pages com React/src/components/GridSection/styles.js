import styled, { css } from 'styled-components';

export const Container = styled.div`
    ${({ theme }) => css``}
`;

export const Grid = styled.div`
    ${({ theme }) => css``}
`;

export const GridElement = styled.div`
    ${({ theme }) => css``}
`;
