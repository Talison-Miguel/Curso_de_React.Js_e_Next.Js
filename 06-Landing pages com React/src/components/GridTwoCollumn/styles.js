import styled, { css } from 'styled-components';

export const Container = styled.div`
    ${({ theme }) => css``}
`;

export const TextContainer = styled.div`
    ${({ theme }) => css``}
`;

export const ImageContainer = styled.div`
    ${({ theme }) => css``}
`;
