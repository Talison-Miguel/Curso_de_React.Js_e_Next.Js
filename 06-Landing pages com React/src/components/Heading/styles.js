import styled, { css } from 'styled-components';

const titleSize = {
    small: (theme) => css`
        font-size: ${theme.font.sizes.medium};
    `,
    medium: (theme) => css`
        font-size: ${theme.font.sizes.large};
    `,
    big: (theme) => css`
        font-size: ${theme.font.sizes.xlarge};
    `,
    huge: (theme) => css`
        font-size: ${theme.font.sizes.xhuge};
        ${mediaFont(theme)}
    `,
};

const mediaFont = (theme) => css`
    @media ${theme.media.lessThanMedium} {
        font-size: ${theme.font.sizes.xlarge};
    }
`;

const titleCase = (uppercase) => css`
    text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
    ${({ theme, colordark, size, uppercase }) => css`
        color: ${colordark ? theme.colors.primaryColor : theme.colors.white};
        ${titleSize[size](theme)}
        ${titleCase(uppercase)}
    `}
`;
