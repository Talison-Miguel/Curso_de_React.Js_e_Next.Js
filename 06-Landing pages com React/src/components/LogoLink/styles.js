import styled, { css } from 'styled-components';

export const Container = styled.a`
    ${({ theme }) => css`
        text-decoration: none;
        color: inherit;
        display: flex;
        align-items: center;

        > img {
            max-height: 3rem;
        }
    `}
`;
