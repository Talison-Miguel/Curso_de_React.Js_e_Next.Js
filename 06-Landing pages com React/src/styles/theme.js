export const theme = {
    colors: {
        primaryColor: '#0a1128',
        secondaryBg: '#dc143c',
        white: '#fff',
    },
    font: {
        family: {
            default: "'Open Sans', sans-serif",
            secondary: "'Montserrat', sans-serif",
        },
        sizes: {
            xmall: '8rem',
            small: '1.6rem',
            medium: '2.4rem',
            large: '3.2rem',
            xlarge: '4.0rem',
            xxlarge: '4.8rem',
            huge: '5.6rem',
            xhuge: '6.4rem',
        },
    },
    media: {
        lessThanMedium: '(max-width: 768px)',
    },
    spacings: {
        xmall: '8rem',
        small: '1.6rem',
        medium: '2.4rem',
        large: '3.2rem',
        xlarge: '4.0rem',
        xxlarge: '4.8rem',
        huge: '5.6rem',
        xhuge: '6.4rem',
    },
};
