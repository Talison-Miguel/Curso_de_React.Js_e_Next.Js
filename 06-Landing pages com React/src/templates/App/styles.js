import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
    ${({ theme }) => css`
        background-color: ${theme.colors.secondaryBg};
    `}
`;
