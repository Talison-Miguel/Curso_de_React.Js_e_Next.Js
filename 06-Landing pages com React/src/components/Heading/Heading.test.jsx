import { screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { Heading } from '.';
import { renderTheme } from '../../styles/renderTheme';
import { theme } from '../../styles/theme';

describe('<Heading />', () => {
    it('should render with default values', () => {
        renderTheme(<Heading>texto</Heading>);
        const heading = screen.getByRole('heading', { name: 'texto' });

        expect(heading).toHaveStyle({
            color: theme.colors.primaryColor,
            'font-size': theme.font.sizes.xhuge,
            'text-transform': 'uppercase',
        });
    });

    it('should render with primaryColor color', () => {
        renderTheme(<Heading colordark={false}>texto</Heading>);
        const heading = screen.getByRole('heading', { name: 'texto' });

        expect(heading).toHaveStyle({
            color: theme.colors.primaryColor,
        });
    });

    it('should render correct render sizes', () => {
        const { rerender } = renderTheme(<Heading size={'small'}>texto</Heading>);
        const heading = screen.getByRole('heading', { name: 'texto' });

        expect(heading).toHaveStyle({
            'font-size': theme.font.sizes.medium,
        });

        rerender(
            <ThemeProvider theme={theme}>
                <Heading size={'big'}>texto</Heading>
            </ThemeProvider>,
        );

        expect(screen.getByRole('heading', { name: 'texto' })).toHaveStyle({
            'font-size': theme.font.sizes.xlarge,
        });

        rerender(
            <ThemeProvider theme={theme}>
                <Heading size={'medium'}>texto</Heading>
            </ThemeProvider>,
        );

        expect(screen.getByRole('heading', { name: 'texto' })).toHaveStyle({
            'font-size': theme.font.sizes.large,
        });

        rerender(
            <ThemeProvider theme={theme}>
                <Heading size={'huge'}>texto</Heading>
            </ThemeProvider>,
        );

        expect(screen.getByRole('heading', { name: 'texto' })).toHaveStyle({
            'font-size': theme.font.sizes.xhuge,
        });
    });

    it('should render correct font sizes when using mobile', () => {
        const { rerender } = renderTheme(<Heading size={'small'}>texto</Heading>);
        const heading = screen.getByRole('heading', { name: 'texto' });

        rerender(
            <ThemeProvider theme={theme}>
                <Heading size={'small'}>texto</Heading>
            </ThemeProvider>,
        );

        expect(screen.getByRole('heading', { name: 'texto' })).toHaveStyle(`font-size: ${theme.font.sizes.medium}`, {
            media: theme.media.lessThanMedium,
        });
    });

    it('should render with uppercase letters', () => {
        renderTheme(<Heading uppercase>texto</Heading>);
        const heading = screen.getByRole('heading', { name: 'texto' });

        expect(heading).toHaveStyle({
            'text-transform': 'uppercase',
        });
    });

    it('should render correct heading element', () => {
        const { container } = renderTheme(<Heading as="h6">texto</Heading>);
        const heading = screen.getByRole('heading', { name: 'texto' });
        const h6 = container.querySelector('h6');

        expect(h6.tagName.toLowerCase()).toBe('h6');
    });
});
