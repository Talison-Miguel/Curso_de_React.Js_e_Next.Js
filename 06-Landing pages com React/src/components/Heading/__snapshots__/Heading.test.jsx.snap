// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> Should match snapshot 1`] = `
.c0 {
  color: #0a1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

@media (max-width: 768px) {
  .c0 {
    font-size: 4.0rem;
  }
}

<h1
  class="c0"
  colordark="true"
  uppercase="false"
>
  texto
</h1>
`;
